clang/lib/Basic/Diagnostic.cpp

```diff
@@ -477,6 +485,100 @@ void DiagnosticsEngine::setSeverityForAll(diag::Flavor Flavor,
       setSeverity(Diag, Map, Loc);
 }
 
+namespace {
+class WarningsSpecialCaseList : public llvm::SpecialCaseList {
```

Comment:
Consider moving this class to a separate file.

Reply:
i'd rather keep it here as I actually want to make sure this stays as an implementation detail of diagnostics, and not used by any other parts of clang.

Reply:
Do we need to keep all implementation details in a single file?
Shouldn't we try to avoid growing huge files so they're easier to read, maintain etc.?
If it's useful somewhere else, someone would just move it and use it I guess.